### ISSUE TITLE
Labor Camp shuttle counts as fleeing the station (On icebox)

### ISSUE THREAD

**Issue by BongoDongo2000** [tgstation/tgstation#57813]
[Round ID]: 15848

So apparently the midround antag head rev role does not check if you're on the station. I was on the labor camp and got it right after I arrived. When I finally earned enough points to go back, the labor camp shuttle ride counted me as fleeing the station.

Given that this happened on Ice Box Station, it may be something special with its wacky z levels.

**Comment by TheBonded** [comment]
Well... the shuttle goes to Z level 12 (the bluespace z level that all shuttles go to when moving) so this would happen on any station, not just icebox. to fix this, just set Z level 12 to count as a station z level. That being said, this may be intentional

**Comment by ATH1909** [comment]
> That being said, this may be intentional, as going on the labour shuttle on any other station would take you to lavaland, which counts as fleeing the station.

Yes, going to lavaland already counts as fleeing the station, so going to the shuttle transit z-level making you count as having fled the station is redundant.

**Comment by TheBonded** [comment]
> > That being said, this may be intentional, as going on the labour shuttle on any other station would take you to lavaland, which counts as fleeing the station.
> 
> Yes, going to lavaland already counts as fleeing the station, so going to the shuttle transit z-level making you count as having fled the station is redundant.

I don't think they have that Z level set to make it count as fleeing, instead they have to set specific z levels to count as not fleeing. For example, on icebox, the second Z wasn't set to be a station Z, which meant going to lower botany counted as fleeing. It seems like non station Zs are automatically considered fleeing, so the mappers don't have to set every other Z level on their map to count as fleeing the station, instead only having to set which z levels are part of the station. Setting the shuttle z level to not count as fleeing would be redundant, as the only times you would be on that z level would be when you are leaving to go to lavaland, which is fleeing, or going to centcom on the emergency shuttle, which cannot happen while a revolution is on station. 

Going to lavaland already counts as fleeing the station, so setting the shuttle transit z-level to not count as fleeing the station would be redundant

**Comment by Whoneedspacee** [comment]
easiest fix might be just connecting it to security map wise and ignoring the shuttle part

**Comment by TheBonded** [comment]
> easiest fix might be just connecting it to security map wise and ignoring the shuttle part

or just setting z level 12 to count as a station z level on every map, wouldn't be hard
or even just on icebox, since there's no other map that the end of the shuttle ride isn't fleeing

**Comment by projectkepler-ru** [comment]
so does anyone have a good solution yet
